Historical Curve Data Plot On Yield Curve Charts [SF-1166]

vijayeth opened the pull request.

as1602 reviewed (commented):
Can you check if these svg's are available from the heroicons package?

as1602 commented:
The storybook has 2 test breaks. Please fix them.

as1602 commented on src/types.ts:
```diff
@@ -108,6 +108,15 @@ export enum HistoricalDataIntervals {
     '1MTH' = '2592000',
 }
 
+export enum HistoricalYieldIntervals {
```
what is the difference between HistoricalDataIntervals and HistoricalYieldIntervals?

vijayeth replied:
`HistoricalDataIntervals` is used for historical widgets and candlestick charts, while `HistoricalYieldIntervals` is used for yield curves.

as1602 commented on src/components/atoms/Checkbox/Checkbox.tsx:
```diff
@@ -8,25 +8,37 @@ export const Checkbox = ({
     label?: string;
     size?: CheckboxSizes;
     disabled?: boolean;
+    isGrayScale?: boolean;
```
1. isGrayScale?
What does this prop mean? We should not be defining such specific props where the context of the prop is unknown from the name.
2. Add tests for all these new changes.

vijayeth replied:
Created new component named `TimeScaleCheckbox` & added tests for all new changes.
resolved in aa240c56bcb6125b83ca9c6bf6791eaa500744eb

as1602 commented on src/components/atoms/Checkbox/Checkbox.tsx:
```diff
@@ -64,7 +80,12 @@ export const Checkbox = ({
+                    className={clsx(
+                        'cursor-pointer whitespace-nowrap capitalize',
+                        isGrayScale
+                            ? 'ml-2 font-secondary text-sm text-neutral-400'
```
please check this change. It breaks the checkbox in the order table panel.

as1602 replied:
you should also define the line height of any text element.
Please refer the Figma and add the correct class.

vijayeth replied:
Made changes for gap and line height for checkbox labels.
resolved in aa240c56bcb6125b83ca9c6bf6791eaa500744eb

as1602 commented on src/hooks/useYieldCurveHistoricalRates/useYieldCurveHistoricalRates.ts:
```diff
@@ -0,0 +1,87 @@
+                ).apr;
+                itayoseMarketIndexSet.add(currentIndex);
+            } else {
+                rate = LoanValue.fromPrice(
```
can you pls explain how this works to get the historical rates?

as1602 commented on src/components/organisms/MultiLineChartTab/MultiLineChartTab.tsx:
```diff
@@ -0,0 +1,378 @@
+            const dataUrl = await domtoimage.toPng(refToUse.current);
+            const link = document.createElement('a');
+            link.href = dataUrl;
+            link.download = 'Yield-Curve.png';
```
Can you consult with @taiga7 for the best name of the downloadable png?

taiga7 replied:
@as1602 
I think the following file name would be good:
{Currency}_YieldCurve_{YYYY-MM-DD_HH-mm-ss}.png
Example: USDC_YieldCurve_2025-03-22_06-31-50.png

Additional note: It is preferable to insert the user’s device time.

vijayeth replied:
Updated the file name for the downloadable png in a833be445a62a8b2f94a6d8dc5bbaf35ae4b3bd9

taiga7 commented:
@vijayeth Even if I hover over a Maturity like MAY2025 on the graph, it reverts to the original point (e.g., APR2025) after three seconds, making it difficult to check the information while hovering.
I would like the hovered state on that Maturity to remain until the mouse moves, without automatically returning to another point.

https://github.com/user-attachments/assets/46873132-e446-42f5-8c50-ee6988ffa081

taiga7 commented:
The bar chart for the frontmost Maturity is not displayed correctly, so please fix it.

<img width="1512" alt="Screenshot 2025-03-25 19 06 35" src="https://github.com/user-attachments/assets/72a30042-805d-4f69-a2db-0e9a0c2c25ed" />
<img width="1512" alt="Screenshot 2025-03-25 19 06 43" src="https://github.com/user-attachments/assets/0d28fd5b-7c75-4b74-a337-87549d5d0538" />
<img width="1512" alt="Screenshot 2025-03-25 19 06 49" src="https://github.com/user-attachments/assets/c7d64da8-dfd4-4179-8e87-e227cca8c31b" />

taiga7 commented:
The highest APR in the graph is cut off and not displayed. (For example, for MAR2025-2 with an APR of 48.11%, the Y-axis only goes up to 46%, causing the graph to be truncated.)
Please adjust the upper limit of the Y-axis so that the highest APR is fully visible on the graph. For an APR of 48.11%, for instance, set the Y-axis limit to a nearby value such as 48.2% or 49% so it is not cut off.
<img width="810" alt="Screenshot 2025-03-25 19 10 28" src="https://github.com/user-attachments/assets/a895f3c2-6a0a-4663-ace9-21a3b145cf8f" />

taiga7 commented:
When you select past data and hover over a specific Maturity on the graph, you can compare current and past values, but the displayed data remains in the order the user selected.
I would like the data to be rearranged so that the current value is first, followed by past data from the most recent to the oldest.
For example, even if the user selects `4H → 1H → 30M`, please display them in the order `Current → 30M → 1H → 4H`.

<img width="1141" alt="Screenshot 2025-03-25 19 34 14" src="https://github.com/user-attachments/assets/8cd50c0f-a71e-428a-8c66-719bc7da4299" />

taiga7 commented:
The position of "Maturity" is shifted to the right relative to the vertical line on the graph, so please center it.
<img width="795" alt="Screenshot 2025-03-26 6 47 32" src="https://github.com/user-attachments/assets/36076d0e-1084-440c-b400-eab5c7c847e0" />

taiga7 commented:
In the attached video, past APR is calculated based on the `Current Mark Price`, but this is incorrect.

For example, the APR from 30 minutes ago must be calculated based on the `Last Mark Price` at that time.

Please note just in case: The `Last Price` is updated regardless of the amount size, so it should not be used for plotting the yield curve.

https://github.com/user-attachments/assets/7d4151fe-8459-423e-a62a-80b63b70c136

vijayeth commented:
@taiga7, I've fixed the tooltip hover behaviour, adjusted the selected timescales with sorting from most recent to oldest in both the label and tooltip, and refined the bar chart styling in 7d38ac54d655d55ec75e8d76c0ca71b1bfde8163

taiga7 commented:
> @taiga7, I've fixed the tooltip hover behaviour, adjusted the selected timescales with sorting from most recent to oldest in both the label and tooltip, and refined the bar chart styling in [7d38ac5](https://github.com/Secured-Finance/secured-finance-app/commit/7d38ac54d655d55ec75e8d76c0ca71b1bfde8163)

It looks great!

Just one small detail—here's some feedback:
As shown in the attached video, if you move your pointer away from the tooltip area and quickly return, it forcefully moves to the Maturity of the currently selected order book.
Is it possible to keep the tooltip displayed when you quickly return the pointer?

If that is difficult, you may ignore this request.

https://github.com/user-attachments/assets/05e55d96-90de-4526-a30f-90aa662c99e0

as1602 commented on src/components/organisms/MultiLineChartTab/MultiLineChartTab.tsx:
```diff
@@ -0,0 +1,358 @@
+    const { currency, maturity } = useSelector((state: RootState) =>
+        selectLandingOrderForm(state.landingOrderForm)
+    );
+    const currentChainId = useSelector(
```
Use chainid from the securedFinance object. This is not always the accurate chainId.
